nix/build-dapp-package.nix

```diff
@@ -0,0 +1,76 @@
+{ pkgs }:
+
+let
+  remappings = xs:
+    builtins.foldl' pkgs.lib.mergeAttrs {}
+      (builtins.map
+        (x: {
+          "${x.name}/" = "${x}/dapp/${x.name}/src/";
+         } // x.remappings)
+         xs);
+  libPaths = xs:
+    builtins.foldl' pkgs.lib.mergeAttrs {}
+      (builtins.map
+        (x: {
+          "${x.name}" = "${x}/dapp/${x.name}/src";
+         } // x.libPaths)
+         xs);
+in
+  {
+    name
+  , src
+  , deps ? []
+  , solc ? pkgs.solc
+  , test-hevm ? pkgs.hevm
+  , solcFlags ? ""
+  , doCheck ? true
+  , ... }@args:
+    pkgs.stdenv.mkDerivation ( rec {
+      inherit doCheck;
+
+      buildInputs = [ test-hevm solc ];
+      passthru = {
+        remappings = remappings deps;
+        libPaths = libPaths deps;
+      };
+
+      REMAPPINGS =
+        pkgs.lib.mapAttrsToList
+          (k: v: k + "=" + v)
+          passthru.remappings;
+
+      LIBSCRIPT =
+        pkgs.lib.mapAttrsToList
+          (k: v: ''
+            ln -s ${v} lib/${k}
+          '')
+          passthru.libPaths;
+
+      buildPhase = args.buildPhase or ''
+        jsonopts=--combined-json=abi,bin,bin-runtime,srcmap,srcmap-runtime,ast
```

Comment:
NOTE: I noticed this is not taking into account the recent changes introduced in #400, fixing this.